superagi/helper/github_helper.py

```diff
@@ -335,3 +337,112 @@ def _get_file_contents(self, file_name, agent_id, agent_execution_id, session):
+        """
+        Gets the content of a specific pull request from a GitHub repository.
+
+        Args:
+            repository_owner (str): Owner of the repository.
+            repository_name (str): Name of the repository.
+            pull_request_number (int): pull request id.
+            headers (dict): Dictionary containing the headers, usually including the Authorization token.
+
+        Returns:
+            dict: Dictionary containing the pull request content or None if not found.
+        """
+        pull_request_url = f'https://api.github.com/repos/{repository_owner}/{repository_name}/pulls/{pull_request_number}'
+        headers = {
+            "Authorization": f"token {self.github_access_token}" if self.github_access_token else None,
+            "Content-Type": "application/vnd.github+json",
+            "Accept": "application/vnd.github.v3.diff",
+        }
+
+        response = requests.get(pull_request_url, headers=headers)
+
+        if response.status_code == 200:
+            logger.info('Successfully fetched pull request content.')
+            return response.text
+        elif response.status_code == 404:
+            logger.warning('Pull request not found.')
+        else:
+            logger.warning('Failed to fetch pull request content: ', response.text)
+
+        return None
+
+    def get_latest_commit_id_of_pull_request(self, repository_owner, repository_name, pull_request_number):
+        url = f'https://api.github.com/repos/{repository_owner}/{repository_name}/pulls/{pull_request_number}/commits'
+        headers = {
+            "Authorization": f"token {self.github_access_token}" if self.github_access_token else None,
+            "Content-Type": "application/json",
+        }
+        response = requests.get(url, headers=headers)
+        if response.status_code == 200:
+            commits = response.json()
+            latest_commit = commits[-1]  # Assuming the last commit is the latest
+            return latest_commit.get('sha')
+        else:
+            logger.warning(f'Failed to fetch commits for pull request: {response.json()["message"]}')
+            return None
+
+
+    def add_line_comment_to_pull_request(self, repository_owner, repository_name, pull_request_number,
+                                         commit_id, file_path, position, comment_body):
+        comments_url = f'https://api.github.com/repos/{repository_owner}/{repository_name}/pulls/{pull_request_number}/comments'
+        headers = {
+            "Authorization": f"token {self.github_access_token}",
+            "Content-Type": "application/json",
+            "Accept": "application/vnd.github.v3+json"
+        }
+        data = {
+            "commit_id": commit_id,
+            "path": file_path,
+            "position": position,
+            "body": comment_body
+        }
+        response = requests.post(comments_url, headers=headers, json=data)
+        if response.status_code == 201:
```

Comment:
here we can avoid using url and headers and make method small by passing them as param and avoid url,headers,data as temp var?